app/controllers/users.rb

```diff
@@ -1,8 +1,26 @@
 get '/signup' do
-  # if user is logged in, redirect, else
   erb :'users/signup'
 end
 
+get '/login' do
+  erb :'users/login'
+end
+
+post '/login' do
+  @user = User.find_by(username: params[:username])
+
+  if @user.nil?
+    @errors = ["Cannot find user #{params[:username]}."]
+    erb :'users/login'
+  elsif @user.password == params[:password]
```

Comment:
At the database level, the field is called password_hash. However, BCrypt allows us to use password as the attribute, and it will read / write from / to password_hash.